### PR TITLE
[List] Add invalidated property

### LINKED ISSUE #1919: [RealmSwift] List missing `invalidated` property
It's available on `RLMArray` (https://realm.io/docs/objc/latest/api/Classes/RLMArray.html#//api/name/invalidated), but not on `List`.

As reported by @AlDrago here: https://github.com/realm/realm-cocoa/issues/1492#issuecomment-101273517.

### PR DESCRIPTION
**Pull request by segiddins**
Closes #1919.

\c @kishikawakatsumi @jpsim

### CODE REVIEW

**Comment by kishikawakatsumi** [comment]
:+1:

**Comment by tgoyne** [comment]
:+1:

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,7 +8,8 @@ x.x.x Release notes (yyyy-MM-dd)
 
 ### Enhancements
 
-* None.
+* Swift: Added `List.invalidated`, which returns if the given `List` is no longer
```
Comment (jpsim) [inline CHANGELOG.md]: Minor point: I like denoting instance methods/properties with `List().invalidated` and class/static methods/properties as `List.invalidated`, but I may be in the minority here.

Comment (segiddins) [inline CHANGELOG.md]: Well, `List.invalidated` returns the curried function, so I disagree. I'd use `List.Self.foo` or `List.Type.foo` for type-level functions.

Comment (jpsim) [inline CHANGELOG.md]: Ok.

**Comment by jpsim** [comment]
:+1:

**Comment by segiddins** [comment]
@jpsim do I need to be worried about that aborted test?

**Comment by jpsim** [comment]
no, I don't know why it was hanging.

**Pull request merged**